### ISSUE TITLE
[Outreachy applications] Startup task: Train and test a classification model

### ISSUE THREAD

**Issue by dzeber** [mozilla/PRESC#2]
This is a good way to get started with the environment and the problem domain. It will also provide the basis for a test case for future work. At a minimum, you should:

- load a dataset from the repo
- train a classification model from [scikit-learn](https://scikit-learn.org/stable/supervised_learning.html)
- compute an [evaluation metric](https://scikit-learn.org/stable/modules/model_evaluation.html) on a held-out test set

Feel free to include any additional steps you feel are relevant or you are interested in trying out, such as:

- basic exploratory analysis of the dataset
- data preprocessing
- hyperparameter tuning

When you start work on this task, please post a comment here indicating which dataset and model you will be working with so that other contributors can avoid duplicating your work.

**Comment by KaairaGupta** [comment]
Hey! 
Can I work on this??

**Comment by tab1tha** [comment]
I'll use the generated.csv dataset.

**Comment by dzeber** [comment]
@KaairaGupta yes! in fact everyone is encouraged to work on this issue first

**Comment by KaairaGupta** [comment]
Okay.

I'll use winequality.csv dataset

**Comment by mhmohona** [comment]
Hello @dzeber! I want to work on vehicles.csv.

**Comment by mhmohona** [comment]
Hello @dzeber! I want to work on defaults.csv.

**Comment by dzeber** [comment]
Sounds good! To be clear, it's also fine to work on the same dataset as someone else, as there are only 5 datasets currently. The goal is just to collect a diversity of models and approaches across these datasets.

**Comment by shreyagupta30** [comment]
Hello @dzeber, I'd like to work on eeg.csv.

**Comment by Clare-Joyce** [comment]
Hello @dzeber, I will work with the winequality.csv

**Comment by elie-wanko** [comment]
Hello @dzeber, I will be working on the **defaults.csv** dataset.

**Comment by BBimie** [comment]
Hello @dzeber, I'll work with the winequality.csv

**Comment by hammedb197** [comment]
Hello @dzeber, I will be working on vehicles.csv.

**Comment by shiza16** [comment]
Hello @dzeber , I'd like to work on vehicle.csv .

**Comment by terminator-4** [comment]
I would like to work on eeg.csv dataset.

**Comment by alberginia** [comment]
I have been working on the generated.cvs dataset with SVMs.

**Comment by Soniyanayak51** [comment]
I would like to work on defaults.csv @dzeber

**Comment by silvererudite** [comment]
I would be working on vehicles.csv  @dzeber

**Comment by blancadesal** [comment]
Hi all! I will be working on the winequality.csv dataset.

**Comment by SanchiMittal** [comment]
I will be working with winequality.csv dataset.

**Comment by aakankshadhurandhar** [comment]
@dzeber I will be working on the default.csv dataset.

**Comment by Bolaji61** [comment]
Hi @dzeber , I will be working on the vehicles.csv dataset.

**Comment by NailaNeena** [comment]
Hello @dzeber, How can we submit the task after completion?

**Comment by dzeber** [comment]
When you are done with this task, please submit a PR following the guidelines listed in the [README](https://github.com/mozilla/PRESC#contribution-guidelines).

**Comment by maxtastu** [comment]
Hello everyone, I am working on the wine quality dataset. I didn't pick a model yet.

**Comment by pratyush-ragini** [comment]
Hi @dzeber , I shall be working on the vehicles.csv dataset

**Comment by NailaNeena** [comment]
Hello @dzeber, I have some question regarding training data-set. I am using Vehicles data-set, I load the dataset. But my question is that should we train it against some rows or against some columns?

**Comment by janvi04** [comment]
Hi @dzeber , I will be working on wine quality dataset.

**Comment by asthad16** [comment]
hello, i am beginning my learning and exploration by working on wine quality dataset.

**Comment by NailaNeena** [comment]
Is there any one who could till me that how to identify the dependent and independent variable in a data-set?
like in Vehicles what will be the dependent variable?

**Comment by Soniyanayak51** [comment]
> Is there any one who could till me that how to identify the dependent and independent variable in a data-set?
> like in Vehicles what will be the dependent variable?

@NailaNeena the label column is given for each dataset in the description. For Vehicles, the dependent/target variable is 'Class'.

**Comment by NailaNeena** [comment]
Thanks @Soniyanayak51

**Comment by simran0117** [comment]
Hey, I am working with wine quality dataset and using SVMs and KNN.

**Comment by asthad16** [comment]
#26

**Comment by asthad16** [comment]
#26

**Comment by brandan1989** [comment]
Hello @dzeber, I will be working on the vehicles.csv dataset.

**Comment by lalapupa** [comment]
Hello @dzeber, I'd like to work on defaults.csv

**Comment by dzekem** [comment]
Hi @dzeber I will be working on the **Defaults.csv dataset.**

**Comment by opeyemiferanmi1** [comment]
Hi @dzeber I will be working on the vehicles.csv dataset

**Comment by shashigharti** [comment]
Hi @dzeber I  will use vehicles.csv dataset

**Comment by NitiKaur** [comment]
hello @dzeber  I will be working with the wine quality dataset and will be using SVM and KNN

**Comment by swatik718** [comment]
Hi @dzeber I'm working on the eeg.csv dataset, going to use an instance based classifier model

**Comment by msmelo** [comment]
Hi everyone!
I'm going to work on the winequality.csv dataset.

**Comment by dzeber** [comment]
> I would be working on vehicles.csv @dzeber are we required to push a PR for this issue?

Yes please.

**Comment by dzeber** [comment]
> Hello @dzeber, I have some question regarding training data-set. I am using Vehicles data-set, I load the dataset. But my question is that should we train it against some rows or against some columns?

@NailaNeena You should be training a model that predicts the `Class` column from the other columns.

**Comment by lauramurphy12** [comment]
Hi everyone! I'm going to work on the winequality.csv dataset

**Comment by namrathagopalabhatla** [comment]
Hey! I have started working on vehicles.csv dataset using Logistic Regression

**Comment by NailaNeena** [comment]
> > Hello @dzeber, I have some question regarding training data-set. I am using Vehicles data-set, I load the dataset. But my question is that should we train it against some rows or against some columns?
> 
> @NailaNeena You should be training a model that predicts the `Class` column from the other columns.

Yes, I am done with this task. Thanks

**Comment by archisha-chandel** [comment]
Hi @dzeber , I am working on winequality.csv with the model -  SGDClassifier

**Comment by urvigodha** [comment]
Hey @dzeber I'll be working on winequality.csv using a Random Forest model

**Comment by msmelo** [comment]
> Hi everyone!
> I'm going to work on the winequality.csv dataset.

@dzeber I'm using a Gradient Boosting model

**Comment by tasbihaasim** [comment]
Hi, I am working on defaults.csv dataset with SGDClassifier.

**Comment by arizzogithub** [comment]
Hi @dzeber , I will be working on eeg dataset.

**Comment by asthad16** [comment]
i have done the modifications requested in the same PR #26. please review it. i m sorry for the delay because i was not well.
